### ISSUE TITLE
User should know easily as who he is logged in

### ISSUE THREAD

**Issue by hvelarde** [plone/Products.CMFPlone#962]
I mentioned this before, but here we go again: it's very important for a user to know all the time who is the one logged in on a site.

This is Plone with the toolbar collapsed:

![selection_009](https://cloud.githubusercontent.com/assets/341393/9822964/d06c4a0e-5899-11e5-8eb1-e2123af4a128.png)

These are various other sites showing, one way or another, using the name, the picture or both, who is logged in:

Facebook:

![selection_007](https://cloud.githubusercontent.com/assets/341393/9822991/fba4d0ec-5899-11e5-8eac-6649e45f5ebd.png)

Google:

![selection_008](https://cloud.githubusercontent.com/assets/341393/9822988/fb47fc82-5899-11e5-95f1-109cda861169.png)

GitHub:

![selection_006](https://cloud.githubusercontent.com/assets/341393/9822990/fb716946-5899-11e5-9cd8-0a2691dc6eb3.png)

And, BTW, all of them use an horizontal toolbar.

**Comment by djay** [comment]
Maybe seem a stupid question but why? What user problem is it solving? Perhaps putting in that context will help the prioritisation process.

**Comment by hvelarde** [comment]
for me is pretty obvious and the use case is in the title; @agnogueira do you want to add something else?

**Comment by djay** [comment]
Question is why is it so important to know who you are? Why demand things happen if you can’t explain why?

**Comment by jensens** [comment]
yes please show the users name, its very important information in many cases.

**Comment by hvelarde** [comment]
@djay before Plone 5 you were able to know the name of the logged in user without any additional action; also, this is the way way the web works: all sites that require you to log in show, in one way or another, who the current user is.

you want a use case? if you share a computer with another user, and you left your session open, then another user can, by mistake, change things using your account; we have to avoid that.

**Comment by maartenkling** [comment]
+1 on name showing

**Comment by ezvirtual** [comment]
+1 on displaying name from me too. I think it fits within a reasonable usability / ux expectation that you know who you're logged in as at a glance (without a click/action).

**Comment by bloodbare** [comment]
How do you expect to show the name also on vertical toolbar when its collapsed ? We should rotate the name to make it feet the collapsed left toolbar?

On horizontal we could add the name when its collapsed, but then we should remove the icon, is it going to be consistent ?

I think that by default the toolbar is expanded in left and top position so the name is shown unless you force to collapse, in my opinion if somebody ask to collapse the toolbar is aware that the login name will be hidden, but we could add on horizontal collapsed toolbar.

**Comment by enzus** [comment]
The problem with the collapsed/expanded modes is that the activation of it is a sort of Easter egg. I found it out by chance myself. So, in the scenario described by Hector, we could have a logged in user who activated the collapsed mode and, at some point, share his computer with another Plone user. The user could not know about the collapsed mode...

Could we not use a small thumbnail of the logged in user or, in absence of it, the initial(s) of the user provided by one of those avatar generators that you see around nowadays?
A thumbnail like the one you see used here on github or smaller should fit the collapsed view.
Not a perfect solution yet but hopefully a good compromise?

**Comment by bloodbare** [comment]
We were thinking hard to see how to make expand button more visible, but any solution lead to other UX problems.

I like the idea of initial(s), I love portrait photo in case it exist with a round border.

**Comment by davilima6** [comment]
+1 on consistently showing portrait + name on expanded toolbars and portraits-only on contracted.

After that, initials would be a great solution in the lack of a customized user photo.

**Comment by hvelarde** [comment]
@bloodbare the option to expand/collapse the toolbar should be on the personal preferences form; please follow https://github.com/plone/Products.CMFPlone/issues/807#issuecomment-139624035

I would prefer to show the user name all the time in the horizontal toolbar; having the photo could be a bonus.

**Comment by enzus** [comment]
@hvelarde I disagree the expande/collapse switcher should be in the personal preferences since, as a user, I might want to change it on the fly depending on the circumstances (small screen, focus mode with less clutter...). If think of many desktop apps like Photoshop, Axure or similar you have a lot of palettes that can be fully displayed, minimised or completely hidden.

Agree that when we have space the user name (or even the full name) would be better. The user's photo or the initial(s) would work as a replacement /not a simple bonus) to give an hint on the identity of the user when there's no space to display the username.

**Comment by davilima6** [comment]
@enzus Having a personal preference doesn't exclude live switching but this discussion should continue in #807

**Comment by tkimnguyen** [comment]
Maybe @hvelarde could rename this issue to indicate that this is only for the horizontal/collapsed toolbar?

**Comment by tkimnguyen** [comment]
OK @hvelarde hold your nose...you know what I'm about to do...

Castle has two toolbars: the usual vertical one on the left and a top bar that separates site settings (a cog or gear icon) and the personal menu, which always shows the full name of the currently logged-in user.

![screen shot 2017-05-08 at may 8 8 56 56 am](https://cloud.githubusercontent.com/assets/815418/25807828/c55e3472-33cd-11e7-9c85-51f29d2fddcf.jpg)
![screen shot 2017-05-08 at may 8 9 07 07 am](https://cloud.githubusercontent.com/assets/815418/25807838/d05e29c2-33cd-11e7-9d4f-2789a834c0a7.jpg)
![screen shot 2017-05-08 at may 8 9 07 15 am](https://cloud.githubusercontent.com/assets/815418/25807839/d261f096-33cd-11e7-9df4-8a668e732131.jpg)

**Comment by hvelarde** [comment]
@tkimnguyen yes, I already knew that; I don't understand why we keep pushing the vertical toolbar when it's obvious is not the right thing for global information and should only be used for local commands (seems Castle still have some issues on that one).

**Comment by jensens** [comment]
this was fixed some time ago ![screenshot_2023-03-13_12:42:45_selection](https://user-images.githubusercontent.com/157140/224692347-860ae51c-f100-4e52-86b2-8f167b3ab434.png)
